core/store/src/trie/mod.rs

```diff
@@ -1711,17 +1712,51 @@ impl Trie {
         &self,
         boundary_account: &AccountId,
         retain_mode: RetainMode,
-        opts: AccessOptions,
-    ) -> Result<StateRoot, StorageError> {
-        let mut trie_update = TrieStorageUpdate::new(&self);
-        let root_node = self.move_node_to_mutable(&mut trie_update, &self.root, opts)?;
-        trie_update.retain_split_shard(boundary_account, retain_mode, opts);
-        #[cfg(test)]
-        {
-            self.memory_usage_verify(&trie_update, GenericNodeOrIndex::Updated(root_node.0));
+    ) -> Result<TrieChanges, StorageError> {
```

Comment:
Where are the changes here coming from?

Reply:
Well, these are simply trie changes on doing the actual retain_split_shard. Earlier we were just not exposing them in this function and calling `Ok(result.new_root)`.

Reply:
Sorry, I meant why did you need to modify this method.